auraed/src/cri/sandbox_cache.rs

```diff
@@ -0,0 +1,83 @@
+ *        Apache 2.0 License Copyright © 2022-2023 The Aurae Authors          *
+ *                                                                            *
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use super::error::{Result, RuntimeServiceError};
+
+use libcontainer::container::Container;
+use std::collections::HashMap;
+
+type Cache = HashMap<String, Container>;
+
+#[derive(Debug, Clone, Default)]
+pub struct SandboxCache {
+    cache: Cache,
+}
+
+impl SandboxCache {
+    pub fn add(
+        &mut self,
+        sandbox_id: String,
+        sandbox: Container,
+    ) -> Result<()> {
+        if self.cache.contains_key(&sandbox_id) {
+            return Err(RuntimeServiceError::SandboxExists { sandbox_id });
+        }
+        let _ = self.cache.insert(sandbox_id, sandbox);
+        Ok(())
+    }
+
+    pub fn get_mut(&mut self, sandbox_id: &String) -> Result<&mut Container> {
```

Comment:
We should eventually refactor to have most logic in the cache if possible and then not have a mut ref exposed.